packages/editor/src/components/block-settings-menu/reusable-block-delete-button.js

```diff
@@ -35,18 +35,27 @@ export default compose( [
 			getBlock,
 			__experimentalGetReusableBlock: getReusableBlock,
 		} = select( 'core/editor' );
+		const { canUser } = select( 'core' );
+
 		const block = getBlock( clientId );
+
+		const reusableBlock = block && isReusableBlock( block ) ?
+			getReusableBlock( block.attributes.ref ) :
+			null;
+
 		return {
-			reusableBlock: block && isReusableBlock( block ) ? getReusableBlock( block.attributes.ref ) : null,
+			isVisible: !! reusableBlock && canUser( 'delete', 'blocks', reusableBlock.id ),
+			isDisabled: reusableBlock && reusableBlock.isTemporary,
 		};
 	} ),
-	withDispatch( ( dispatch, { onToggle = noop } ) => {
+	withDispatch( ( dispatch, { clientId, onToggle = noop }, { select } ) => {
```

Comment:
`select` - that's interesting to see it have much broader application than I anticipated. I hope there are no drawbacks of that approach :)

Reply:
I like that in this case it lets us avoid passing unnecessary props to the component.